src/filter/aech0r/aech0r.cpp

```diff
@@ -0,0 +1,308 @@
+// TODO RGB gradiant by fading influence need more love (See '//Fade by color layers')!
+// FIXME SSE2 version doesnt support RGB fading influence !
+
+// FIXME (Veejay specifics?) on activate/desactivate/activate/..., some buffers must be cleared!
+
+// EXPORE ME -
+//~ if((skip_count++)>m_skip) {
+//~ -      skip_count = 0;
+//~ -      m_factor += (factor * 64);
+//~ [...]
+//~ }
+
+// EXPLORE ME a very high m_factor value give some interresting color result (      m_factor += (factor * 64) * m_skip;)
+
+
+/* Intrinsic declarations */
+#if defined(__SSE2__)
+#include <emmintrin.h>
+// TODO mmx support and others
+//  #elif defined(__MMX__)
+// #include <mmintrin.h>
+#endif
+
+#define SKIP_MAX_IMAGES 8
+#define M_FACTOR_MAV    127
+
+union px_t {
+  uint32_t u;
+  unsigned char c[4]; // 0=B, 1=G,2=R,3=A ? i think :P
+};
+
+class aech0r : public frei0r::filter {
+private:
+  f0r_param_double factor;
+  //~ f0r_param_double factor_r;  //Fade by color layers
+  //~ f0r_param_double factor_g;
+  //~ f0r_param_double factor_b;
+  f0r_param_double strobe_period;
+  bool bright;
+  bool flag_r;
+  bool flag_g;
+  bool flag_b;
+
+
+  //~ f0r_param_double fade_rgb;  //Fade by color layers
+  //~ f0r_param_double flag_rgb;
+
+  unsigned int m_factor;
+
+  //~ unsigned int m_factor_r;  //Fade by color layers
+  //~ unsigned int m_factor_g;
+  //~ unsigned int m_factor_b;
+
+  //~ unsigned int m_flag_rgb;  //Fade by color layers
+  //~ unsigned int m_flag_r;
+  //~ unsigned int m_flag_g;
+  //~ unsigned int m_flag_b;
+
+  unsigned int m_skip;
+  unsigned int m_skip_count;
+
+  bool firsttime;
+
+  //~ unsigned int m_rgb; //Fade by color layers
+
+#ifdef __SSE2__
+  long long int m_factor_sse2;
+
+  inline void tracesse2_add(uint32_t* out, const uint32_t* in);
+  inline void tracesse2_sub(uint32_t* out, const uint32_t* in);
+#else
+  unsigned int m_factor_r;
+  unsigned int m_factor_g;
+  unsigned int m_factor_b;
+
+  inline void trace_add(uint32_t* out, const uint32_t* in);
+  inline void trace_sub(uint32_t* out, const uint32_t* in);
+#endif
+public:
+
+  aech0r(unsigned int width, unsigned int height) {
+
+    factor = 0.15; // Quasi full echo has default
+    bright = false; // Dark mode has default
+    
+    flag_r = false; // No RGB flag has default
+    flag_g = false;
+    flag_b = false;
+
+    strobe_period = 0; // No strobe has default
+
+    //~ fade_rgb = ?  //Fade by color layers
+
+    firsttime = true;
+    m_skip_count = 0;
+
+    register_param(factor, "Fade Factor", "Disappearance rate of the echo"); // 0 No fade, 1 No Trace
+    register_param(bright, "Direction", "Darker or Brighter echo"); // Add or Substract data
+    register_param(flag_r, "Keep RED", "Influence on Red canal"); // 0 Fade canal, 1 Keep canal data
+    register_param(flag_g, "Keep GREEN", "Influence on Green canal"); // 0 Fade canal, 1 Keep canal data
+    register_param(flag_b, "Keep BLUE", "Influence on Blue canal"); // 0 Fade canal, 1 Keep canal data
```

Comment:
@d-j-a-y If I understood this correctly, these should be spelled as **channel** instead?

```suggestion
    register_param(flag_r, "Keep RED", "Influence on Red channel"); // 0 Fade canal, 1 Keep canal data
    register_param(flag_g, "Keep GREEN", "Influence on Green channel"); // 0 Fade canal, 1 Keep canal data
    register_param(flag_b, "Keep BLUE", "Influence on Blue channel"); // 0 Fade canal, 1 Keep canal data
```

See [Channel (digital image)](https://en.wikipedia.org/wiki/Channel_(digital_image)) on Wikipedia:

> a **channel** is any grayscale image of the same dimension as and associated with the original image.